static/src/components/VacancyView.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable react/prop-types */
+import { Box, Grid, Typography } from "@material-ui/core";
+import React, { useEffect, useState, Fragment } from "react";
+import { useParams } from "react-router-dom";
+import LinkButton from "./LinkButton";
+
+export default function VacancyView(props) {
+    const { id } = useParams();
+    const [vacancy, setVacancy] = useState({});
+
+    useEffect(() => {
+        const [choosenVacancy] = [
+            ...props.vacancies.filter((vacancy) => parseInt(id) === vacancy.id),
+        ];
+        setVacancy(choosenVacancy);
+    }, []);
+
+    return (
+        <Fragment>
+            <Grid container justify="space-between" alignItems="flex-end">
+                <Grid item>
+                    <Typography variant="h6">{vacancy?.position}</Typography>
+                </Grid>
+                <Grid item>
+                    <Typography>{vacancy?.salary} ₽ / месяц</Typography>
+                </Grid>
```

Comment:
здесь и в других похожих местах лучше написать как-то вот так 
```suggestion
                {vacancy.salary && <Grid item>
                    <Typography>{vacancy.salary} ₽ / месяц</Typography>
                </Grid>}
```

Reply:
то есть вообще не рендерить Grid полностью, если salary нет, иначе у тебя будет пустой ненужный элемент